cpp/common/src/codingstandards/cpp/Literals.qll

```diff
@@ -28,3 +28,22 @@ class Utf16StringLiteral extends StringLiteral {
 class Utf32StringLiteral extends StringLiteral {
   Utf32StringLiteral() { this.getValueText().regexpMatch("(?s)\\s*U\".*") }
 }
+
+/**
+ * A literal resulting from the use of a constexpr
+ * variable, or macro expansion.
+ */
+class CompileTimeComputedIntegralLiteral extends Literal {
```

Comment:
hm was looking at a (I think) similarly [concepted CodeQL query](https://github.com/github/codeql/blob/main/cpp/ql/src/Best%20Practices/Magic%20Constants/MagicConstants.qll#L75C45-L75C59) - should we be caring about floating point types also?

Reply:
Not at this point. Floating points cannot be used as [template non type parameters](https://en.cppreference.com/w/cpp/language/template_parameters) for C++ 14. This has been added for C++ 20 so I will add it as a comment.

Reply:
The other use cases, where we have `#define delta 0.1f` is already covered by the query excluding macro calls.